GameEngine/Agents/SearchNode.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+
+namespace GameEngine.Agents
+{
+    public class SearchNode
+    {
+        public GameState State { get; protected set; }
+        public SearchNode Parent { get; protected set; }
+        public Play Action { get; protected set; }
+        public int PathCost { get; protected set; }
+        public int StepCost { get { return 1; } }
```

Comment:
💡 Private?

Reply:
If we keep the same heuristic (see @DrSeabass comments above), then I don't see where PathCost or StepCost are used. Are these future-proofing for later algorithms which will use them?

Reply:
They are vaguely future-proofing. I mostly included them for compatibility with the literature (which will probably matter for future work). It also seemed like good documentation as code since it encapsulates the knowledge (the decision?) that all step costs are 1. The example I followed was the implementation of `ChildNode` given in section 3.3.1 of _A Modern Approach_ (3rd ed.). The only changes were to move to more of an OO than a functional style.

On the other hand as you say, we're really not using `PathCost` yet. Maybe we won't. I don't really have a problem with removing for now. YAGNI and all that. Especially if we anticipate that future heuristic and cost functions will measure in something other than number of (non-sun) plays. In other words, I think `StepCost` will always be 1. I'm having a hard time coming up with any other measure that makes sense. @DrSeabass , do you have any thoughts on this?

Reply:
I'm keeping it as very likely useful. I've made it a property of a `Play` that always returns 1.